### PR TITLE
dep!(dev): bump eslint w/ corrections

### PR DESCRIPTION
**Pull request by erisu**
### Motivation and Context

Update Eslint Config with Fixes

### Description

- `@cordova/eslint-config@^4.0.0`
- applied auto corrections
- applied manual corrections
  - prefer-regex-literals

### Testing

- `npm t`

### Checklist

- [x] I've run the tests to see all new and existing tests pass

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -24,3 +24,5 @@ overrides:
 
   - files: [cordova-js-src/**/*.js]
     extends: '@cordova/eslint-config/browser'
+    rules:
+      no-var: 0
```
Comment (erisu) [inline .eslintrc.yml:28]: This rule should be applied directly to `@cordova/eslint-config/browser` but in the mean time applied here. Will hope to reoslve this in a later patch release of  `@cordova/eslint-config`.

**Review by raphinesse — APPROVED** [review]
I did some commit squashing and added a TODO about the rule exception. Good to merge from my side.

**Review by raphinesse — CHANGES_REQUESTED** [review]
Sorry, I messed up the lockfile :(
I'll fix that!

**Review by raphinesse — APPROVED** [review]
OK, all good now

**Comment by raphinesse** [comment]
I just remembered that we only enforce ES5 for browser code that lives in `cordova-js`. We could actually migrate the `cordova-js-src` that lives in this repo to ES6 (or beyond) **if** we can guarantee that it is supported by the iOS WebView versions we support. Is that the case?

So I think we should not disable the `no-var` rule in `@cordova/eslint-config`, since a platform should be able to decide whether it needs to support ES5-only environments or not.

Nevertheless this PR is still good to go for me (maybe remove my TODO if you agree with the above).

**Pull request merged**